lib/cc/pull_requests.rb

```diff
@@ -1,8 +1,28 @@
 class CC::PullRequests < CC::Service
+  ABLE_TO_UPDATE_STATUS_MESSAGE = "Access token is valid"
+  ABLE_TO_UPDATE_STATUS_AND_COMMENT_MESSAGE = "Access token is valid"
+  ABLE_TO_UPDATE_STATUS_BUT_NOT_COMMENT_MESSAGE = "OAuth token requires 'repo' scope to post comments."
   def receive_test
     setup_http
 
-    receive_test_status
+    # this will raise an HTTPError or be successful:
+    able_to_update_status_response = check_if_able_to_update_status
+
+    if welcome_comment_implemented? && config.welcome_comment_enabled
```

Comment:
If we want to make them more similar, I think we could make the status check more like the comment check. The comment check makes a request to confirm that the provided auth token has been granted the appropriate permissions which it will need to post a comment. The status check, on the other hand, attempts to update a status for a nonexistent commit (one with a SHA of `0000000000000000000000000000000000000000`) and checks the status code of the response to make sure it indicates "you're allowed to, but that commit doesn't exist" rather than "you're not allowed to". The downside to this approach is that, I believe, it will retry 3 times (which is a bit slow/futile) and doesn't give as actionable advice (telling them which permission was missing from the oauth token). Changing this behavior would mean changing both the github and the gitlab integration. At this point in time I'm inclined to tolerate the discrepancy but will get another opinion on it